src/main/java/bio/terra/service/filedata/azure/util/BlobContainerCopier.java

```diff
@@ -0,0 +1,225 @@
+   * performs the copy operation. The method returns an instance of the {@link
+   * BlobContainerCopySyncPoller} which facilitates the management and getting the status of the
+   * copy operation.
+   *
+   * @return an instance of {@link BlobContainerCopySyncPoller} that is associated with the copy
+   *     operation.
+   */
+  public BlobContainerCopySyncPoller beginCopyOperation() {
+
+    if (this.sourceClientFactory != null) {
+      logger.info("Starting copy operation using a container as a source");
+      return beginCopyOperationUsingSourceBlobContainerClient();
+    }
+
+    if (StringUtils.isNotBlank(this.sourceBlobUrl)) {
+      logger.info("Starting copy operation using a signed blob URL as the source");
+      return beginCopyOperationUsingBlobSignedUrl();
+    }
+
+    throw new IllegalArgumentException(
+        "The copy operation can't be started. " + "The source information is missing or invalid");
+  }
+
+  public void setDestinationBlobName(String destinationBlobName) {
+    this.destinationBlobName = destinationBlobName;
+  }
+
+  public void setSourceDestinationPairs(
+      List<BlobCopySourceDestinationPair> sourceDestinationPairs) {
+    this.sourceDestinationPairs = sourceDestinationPairs;
+  }
+
+  public void setSourceBlobUrl(String sourceBlobUrl) {
+    this.sourceBlobUrl = sourceBlobUrl;
+  }
+
+  public void setBlobSourcePrefix(String blobSourcePrefix) {
+    this.blobSourcePrefix =
+        Objects.requireNonNull(blobSourcePrefix, "Blob source prefix must not be null");
+  }
+
+  public void setListOperationTimeout(Duration listOperationTimeout) {
+    this.listOperationTimeout = listOperationTimeout;
+  }
+
+  public void setPollingInterval(Duration pollingInterval) {
+    this.pollingInterval = pollingInterval;
+  }
+
+  public void setSourceClientFactory(BlobContainerClientFactory sourceClientFactory) {
+    this.sourceClientFactory = sourceClientFactory;
+  }
+
+  public BlobContainerClientFactory getDestinationClientFactory() {
+    return destinationClientFactory;
+  }
+
+  public String getBlobSourcePrefix() {
+    return blobSourcePrefix;
+  }
+
+  public BlobContainerClientFactory getSourceClientFactory() {
+    return sourceClientFactory;
+  }
+
+  public String getSourceBlobUrl() {
+    return sourceBlobUrl;
+  }
+
+  public String getDestinationBlobName() {
+    return destinationBlobName;
+  }
+
+  public List<BlobCopySourceDestinationPair> getSourceDestinationPairs() {
+    return sourceDestinationPairs;
+  }
+
+  private BlobContainerCopySyncPoller beginCopyOperationUsingBlobSignedUrl() {
+    BlobUrlParts blobUrlParts = BlobUrlParts.parse(this.sourceBlobUrl);
+
+    String sourceBlobName = blobUrlParts.getBlobName();
+    String destinationBlobName = this.destinationBlobName;
+    if (StringUtils.isBlank(this.destinationBlobName)) {
+      destinationBlobName = sourceBlobName;
+    }
+
+    return new BlobContainerCopySyncPoller(
+        Collections.singletonList(
```

Comment:
In JDK 9+, `List.of(x)` is preferred over `Collections.singletonList(x)`.
```suggestion
        List.of(
```

Reply:
Thanks fixed.